ci_framework/roles/cifmw_cephadm/tasks/dashboard/dashboard.yml

```diff
@@ -0,0 +1,109 @@
+      changed_when: false
+    - name: "set the dashboard SSL port ({{ cifmw_cephadm_dashboard_port }})"
+      command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl_server_port {{ cifmw_cephadm_dashboard_port }}"
+      run_once: true
+
+- name: disable SSL for dashboard
+  become: true
+  command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl false"
+  when: cifmw_cephadm_dashboard_protocol == "http"
+
+- name: Configure SSL key/cert for the Ceph Dashboard if provided
+  become: true
+  block:
+    - name: enable SSL for dashboard
+      command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl true"
+      run_once: true
+
+    - name: import dashboard certificate file
+      command: "{{ cifmw_cephadm_ceph_cli }} config-key set mgr/dashboard/crt -i {{ cifmw_cephadm_dashboard_crt }}"
+      changed_when: false
+
+    - name: import dashboard certificate key
+      command: "{{ cifmw_cephadm_ceph_cli }} config-key set mgr/dashboard/key -i {{ cifmw_cephadm_dashboard_key }}"
+      changed_when: false
+  when: cifmw_cephadm_dashboard_protocol == "https" and
+        cifmw_cephadm_dashboard_crt | length > 0 and
+        cifmw_cephadm_dashboard_key | length > 0
+
+- include_tasks: configure_dashboard_backends.yml
+  with_items: "{{ groups['edpm'] }}"
+  vars:
+    dashboard_backend: '{{ item }}'
+
+- name: Restart the Ceph dashboard
+  become: true
+  block:
+    - name: disable mgr dashboard module (restart)
+      command: "{{ cifmw_cephadm_ceph_cli }} mgr module disable dashboard"
+    - name: enable mgr dashboard module (restart)
+      command: "{{ cifmw_cephadm_ceph_cli }} mgr module enable dashboard"
+
+      #ToDo : skipped dashboard admin user creation task for CI, we shall add it if needed. 
+- name: Configure Monitoring Stack
+  become: true
+  block:
+    - name: get Grafana instance(s) addresses
+      set_fact:
+        grafana_server_addrs: "{{ (grafana_server_addrs | default([])) | union(hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_rgw_network)) }}"
+      loop: "{{ groups['edpm'] | list }}"
+    - name: set grafana api user
+      command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-username {{ cifmw_cephadm_grafana_admin_user }}"
+    - name: set grafana api password
+      command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-password -i -"
+      args:
+        stdin: "{{ cifmw_cephadm_grafana_admin_password }}"
+        stdin_add_newline: no
+    - name: disable ssl verification for grafana
+      command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-ssl-verify False"
+      changed_when: false
+      when:
+        - cifmw_cephadm_dashboard_protocol == "https"
+    - name: set alertmanager host
+      command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-alertmanager-api-host http://{{ grafana_server_addrs | first }}:9093
```

Comment:
my observation during testing is that , even after fetching the ips from only first node, i still see two ips
        "172.18.0.102",
        "172.18.0.2"

iiuc, second one is the vip
so i think we still need '| first' so i didn't update.
what if we don't get the ips in the same order ? some improvement may be need here in next iteration.

Reply:
ack `| first` actually doesn't hurt and prevents failures only because you have `>1` ip address.
My point here was different: if we get (elsewhere) `grafana_addr` as a single variable containing the `IP` address on the storage network where grafana is deployed, you don't need to have `| first` here beacause that value is supposed to be gathered/validates/set beforehand (e.g. in the `ceph.yml` playbook).
This is only a suggestion, not an actual issue, but I think that `IP` already available before and can be injected here as an input.